ndmg/stats/qa_tensor.py

```diff
@@ -33,86 +34,167 @@
+    
+    Parameters
+    -----------
+    peak_dirs: np array of peak_dirs 
+    peak_values: np array of peak_values
+    output_dir: location to save qa figure
+    model: model type (CSA, CSD)
+    """
+    #set shape of image
+    im_shape = peak_dirs.shape[0:3]
+    im_shape_rgb = im_shape + (3,)
+    max_dim = max(im_shape) 
+
+    #title
+    title = f'QA for Tractography {model.upper()} Model Peak Directions. Brain Volume: {im_shape}'
+    
+    #generate 3-d directional data
+    centers,directions,directions_colors,heights = generate_3_d_directions(peak_dirs,peak_values)
+    
+    #reshape back into a 3-d volume with voxel encoded RGB values corresponding to directional vectors
+    im = directions_colors.reshape(im_shape_rgb)
+    
+    x = [int(im_shape[0] * 0.35), int(im_shape[0] * 0.51), int(im_shape[0] * 0.65)]
+    y = [int(im_shape[1] * 0.35), int(im_shape[1] * 0.51), int(im_shape[1] * 0.65)]
+    z = [int(im_shape[2] * 0.35), int(im_shape[2] * 0.51), int(im_shape[2] * 0.65)]
 
-    if im.shape == (182, 218, 182):
-        x = [78, 90, 100]
-        y = [82, 107, 142]
-        z = [88, 103, 107]
-    else:
-        shap = im.shape
-        x = [int(shap[0] * 0.35), int(shap[0] * 0.51), int(shap[0] * 0.65)]
-        y = [int(shap[1] * 0.35), int(shap[1] * 0.51), int(shap[1] * 0.65)]
-        z = [int(shap[2] * 0.35), int(shap[2] * 0.51), int(shap[2] * 0.65)]
     coords = (x, y, z)
-
     labs = [
-        "Sagittal Slice (YZ fixed)",
-        "Coronal Slice (XZ fixed)",
-        "Axial Slice (XY fixed)",
+        "Sagittal Slice",
+        "Coronal Slice",
+        "Axial Slice",
     ]
     var = ["X", "Y", "Z"]
-
+    
     idx = 0
     for i, coord in enumerate(coords):
         for pos in coord:
             idx += 1
             ax = plt.subplot(3, 3, idx)
             ax.set_title(var[i] + " = " + str(pos))
             if i == 0:
-                image = ndimage.rotate(im[pos, :, :], 90)
+                image = ndimage.rotate(im[pos, :, : , :], 90)
             elif i == 1:
-                image = ndimage.rotate(im[:, pos, :], 90)
+                image = ndimage.rotate(im[:, pos, : , :], 90)
             else:
-                image = im[:, :, pos]
-
+                image = im[:, :, pos, :]
             if idx % 3 == 1:
                 ax.set_ylabel(labs[i])
-                ax.yaxis.set_ticks([0, image.shape[0] / 2, image.shape[0] - 1])
-                ax.xaxis.set_ticks([0, image.shape[1] / 2, image.shape[1] - 1])
-
+            #remove axis
+            plt.xticks([])
+            plt.yticks([])
+            image = (image*255).astype(np.uint8)
+            #convert background
+            image = np.where(image<=0.01, 255, image)
+            #pad image size
+            image = pad_im(image,max_dim,255)
             plt.imshow(image)
 
+            
     fig = plt.gcf()
+    fig.suptitle(title)
     fig.set_size_inches(12.5, 10.5, forward=True)
-    return fig
+    fig.savefig(output_dir)
```

Comment:
big problem -- here's what I got when I ran `fig.savefig(output_dir)`:
https://www.dropbox.com/s/2olljaoam1ap7bm/Screenshot%202019-12-09%2011.43.56.png?dl=0

Reply:
Can you elaborate: what parameters and data did you run it on? In my testing it sliced it appropriately. 

I'm guessing either the slicing function or directions failed for a certain data or the values generated were so small my background white conversion steps made it all white.

Reply:
Didn't complete the QA today either.

I'm testing on `sub-0025427`:
https://www.dropbox.com/s/stdvvja038lpdjb/sub-0025427.zip?dl=0

here are my `launch.json` parameters:
```
        "--participant_label",
        "0025427",
        "--session_label",
        "1",
        "--parcellation",
        "desikan",
        "/Users/alex/.ndmg_data/input",
        "/Users/alex/.ndmg_data/output"
```

Here are the local variables inside my debugger:
https://www.dropbox.com/s/2fcr1fm7fmf1qs3/Screenshot%202019-12-10%2010.50.04.png?dl=0

Reply:
I ran this exact data locally and it worked fine.